### ISSUE TITLE
vim.lsp.format clears all the marks 

### ISSUE THREAD

**Issue by dmtrKovalenko** [chentoast/marks.nvim#105]
I use vim lsp format on the simple markdown file and whenever it is called all the marks are lost. Do you know any reason it might happen seems like some kind of bug in the plugin to me, would be happy to help investigating it

https://github.com/chentoast/marks.nvim/assets/16926049/69646a55-10b2-492f-9b9b-034db830a167

**Comment by swaroopanand10** [comment]
Hi, I am facing this same issue, Did you found some solution or workaround for this?

**Comment by dmtrKovalenko** [comment]
Yes, I used https://github.com/stevearc/conform.nvim for formatting which edits only specific parts and specifically saves marks m

**Comment by swaroopanand10** [comment]
@dmtrKovalenko Thanks for the solution, its working.

**Comment by chentoast** [comment]
the main reason for this is because `vim.lsp.buf.format` (under the hood) deletes the entire buffer, and replaces it with new text. As such, there's no way for the plugin to retain marks - and the solution is to use workarounds like conform.nvim - or submit a PR to modify `buf_set_lines` in the neovim source code.
